src/main/java/eventprocessing/models/ReadingAggregator.java

```diff
@@ -0,0 +1,93 @@
+package eventprocessing.models;
+
+import com.google.common.collect.Iterables;
+import eventprocessing.fileservices.CSVFileService;
+
+import java.time.Clock;
+import java.util.ArrayList;
+import java.util.List;
+
+public class ReadingAggregator {
+    private CSVFileService csvFileService;
+    private Clock clock;
+    private List<Bucket> buckets;
+    private int duplicateCounter = 0;
+
+    public int getDuplicateCounter() {
+        return duplicateCounter;
+    }
+
+    public List<Bucket> getBuckets() {
+        return buckets;
+    }
+
+    public ReadingAggregator(CSVFileService csvFileService, Clock clock) {
+        this.csvFileService = csvFileService;
+        this.clock = clock;
+        this.buckets = new ArrayList<>();
```

Comment:
Very minor suggestion: if you move this to line 13, you could make `buckets` final.